timesketch/api/v1/resources_test.py

```diff
@@ -323,6 +323,282 @@ def test_invalid_index(self):
+        self.assertIn(b"Attributes for event exceeds maximum", response.data)
+
+        response = self.client.post(
+            self.resource_url,
+            json={
+                "events": [
+                    {
+                        "_id": "1",
+                        "_index": "1",
+                        "_type": "_doc",
+                        "attributes": [
+                            {"attr_name": "abc"}
+                        ]
+                    }
+                ]
+            }
+        )
+        self.assertEqual(HTTP_STATUS_CODE_BAD_REQUEST, response.status_code)
+        self.assertIn(b"Attribute missing field", response.data)
+
+    @mock.patch(
+        "timesketch.api.v1.resources.OpenSearchDataStore", MockDataStore
+    )
+    def test_chunk_calculation(self):
+        """Tests that chunks are properly calculated."""
+        self.login()
+
+        attrs = [{"attr_name": "foo", "attr_value": "bar"}]
+
+        # One chunk when event count is the same as chunk size.
+        response = self.client.post(
+            self.resource_url,
+            json={
+                "events": [
+                    {
+                        "_id": "1",
+                        "_type":"_doc",
+                        "_index": "1",
+                        "attributes": attrs
+                    }
+                ]*1000
+            }
+        )
+        self.assertEqual({"1":1}, response.json["meta"]["chunks_per_index"])
+
+        # Two chunks when event count is chunk size + 1.
+        response = self.client.post(
+            self.resource_url,
+            json={
+                "events": [
+                    {
+                        "_id": "1",
+                        "_type":"_doc",
+                        "_index": "1",
+                        "attributes": attrs
+                    }
+                ]*1001
+            }
+        )
+        self.assertEqual({"1":2}, response.json["meta"]["chunks_per_index"])
+
+        # Chunk per index with multiple indexes.
+        response = self.client.post(
+            self.resource_url,
+            json={
+                "events": [
+                    {
+                        "_id": "1",
+                        "_type":"_doc",
+                        "_index": "1",
+                        "attributes": attrs
+                    },
+                    {
+                        "_id": "1",
+                        "_type":"_doc",
+                        "_index": "2",
+                        "attributes": attrs
+                    }
+                ]
+            }
+        )
+        self.assertEqual(
+            {"1":1, "2":1}, response.json["meta"]["chunks_per_index"])
+
+    @mock.patch(
+        "timesketch.api.v1.resources.OpenSearchDataStore", MockDataStore
+    )
+    def test_add_invalid_attributes(self):
+        """Tests existing attributes cannot be overidden."""
```

Comment:
again rather split it up in smaller individual tests

Reply:
Done